src/main/java/com/microservices/springkafkaconsumer/listener/OrderNotificationListener.java

```diff
@@ -6,46 +6,40 @@
 import com.microservices.springkafkaconsumer.bean.User;
 import com.microservices.springkafkaconsumer.repo.OrderCRUD;
 import com.microservices.springkafkaconsumer.repo.UserCRUD;
+import lombok.extern.slf4j.Slf4j;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.beans.factory.annotation.Value;
 import org.springframework.kafka.annotation.KafkaListener;
 import org.springframework.stereotype.Service;
 
+import java.util.Optional;
+
 @Service
+@Slf4j
 public class OrderNotificationListener {
 
-    @Value("${order.topic.name}")
-    private String topicName;
+
 
     @Autowired
     OrderCRUD orderCRUD;
     @Autowired
     UserCRUD userCRUD;
 
-    @KafkaListener(topics = "order-topic", groupId = "foo")
-    public void listenGroupFoo(String message) {
+
+    @KafkaListener(topics = "ORDER_TOPIC", groupId = "foo")
+    public void listenGroupFoo(String message) throws JsonProcessingException {
```

Comment:
its more appropriate here if you put the block between try/catch to handle the exception